python/altrios/demos/speed_limit_train_sim_demo.py

```diff
@@ -151,9 +149,62 @@
 ax[-1].set_ylabel('Speed [m/s]')
 ax[-1].legend()
 plt.suptitle("Speed Limit Train Sim Demo")
+
+fig2, ax1 = plt.subplots(3, 1, sharex=True)
+ax1[0].plot(
+    np.array(train_sim.history.time_seconds) / 3_600,
+    np.array(train_sim.history.offset_in_link_meters) / 1_000,
+)
+ax1[0].set_ylabel('Offset in Current\nLink [km]')
+
+ax1[1].plot(
+    np.array(train_sim.history.time_seconds) / 3_600,
+    train_sim.history.head_end_link_idx,
+)
+ax1[1].set_ylabel('Head End\nLink Index')
+
+ax1[-1].plot(
+    np.array(train_sim.history.time_seconds) / 3_600,
+    train_sim.history.speed_meters_per_second,
+)
+ax1[-1].set_xlabel('Time [hr]')
+ax1[-1].set_ylabel('Speed [m/s]')
+
+plt.suptitle("Speed Limit Train Sim Demo")
+plt.tight_layout()
+
+
+fig1, ax2 = plt.subplots(3, 1, sharex=True)
```

Comment:
<img width="652" alt="image" src="https://github.com/NREL/altrios/assets/854536/027bf5be-5021-4efc-9ae0-6447a8a128f8">

This demonstrates how to plot grade at front of train